kolibri/core/analytics/utils.py

```diff
@@ -62,6 +73,133 @@
+            queryset
+            # convert birth year to integer
+            .annotate(
+                birth_year_int=Cast("birth_year", output_field=IntegerField())
+            ).aggregate(
+                # count of male users
+                num_males=Sum(
+                    Case(
+                        When(gender=demographics.MALE, then=1),
+                        output_field=IntegerField(),
+                    )
+                ),
+                # count of female users
+                num_females=Sum(
+                    Case(
+                        When(gender=demographics.FEMALE, then=1),
+                        output_field=IntegerField(),
+                    )
+                ),
+                # count of not specified users
+                num_not_specified_gender=Sum(
+                    Case(
+                        When(gender=demographics.NOT_SPECIFIED, then=1),
+                        output_field=IntegerField(),
+                    )
+                ),
+                # count of deferred users
+                num_deferred_gender=Sum(
+                    Case(
+                        When(Q(gender=demographics.DEFERRED) | Q(gender=""), then=1),
+                        output_field=IntegerField(),
+                    )
+                ),
+                # number of users who specified birth year
+                total_specified_birth_year=Sum(
+                    Case(
+                        When(
+                            ~Q(birth_year="")
+                            & ~Q(birth_year=demographics.DEFERRED)
+                            & ~Q(birth_year=demographics.NOT_SPECIFIED),
+                            then=1,
+                        ),
+                        output_field=IntegerField(),
+                    )
+                ),
+                # number of users who did NOT specify birth year
+                num_deferred_birth_year=Sum(
+                    Case(
+                        When(
+                            Q(birth_year="") | Q(birth_year=demographics.DEFERRED),
+                            then=1,
+                        ),
+                        output_field=IntegerField(),
+                    )
+                ),
+                # average birth year of users
+                average_birth_year=Avg(
+                    Case(
+                        When(
+                            ~Q(birth_year="")
+                            & ~Q(birth_year=demographics.DEFERRED)
+                            & ~Q(birth_year=demographics.NOT_SPECIFIED),
+                            then=F("birth_year_int"),
+                        )
+                    ),
+                    output_field=FloatField(),
+                ),
+            )
+        )
+        # payload of birth year and gender statistics
+        stats = {
+            "bys": {
+                "a": demographic_stats["average_birth_year"],
+                "v": variance(list_of_birth_years),
```

Comment:
have a question that's not related to the code implementation - just wondering why we decided to use variance instead of standard deviation here. thank you!

Reply:
It was chosen arbitrarily. Reporting standard deviation instead might be preferable because it's easier to interpret (same unites as mean/average)

@rtibbles @ralphiee22 perhaps switch to SD?

Reply:
SD seems fine to me.

Reply:
thank you for the explanation!